xmake/rules/c++/modules/xmake.lua

```diff
@@ -21,45 +21,122 @@
 -- define rule: c++.build.modules
 rule("c++.build.modules")
     set_extensions(".mpp", ".mxx", ".cppm", ".ixx")
+    add_deps("c++.build.modules.dependencies")
+    add_deps("c++.build.modules.builder")
+    add_deps("c++.build.modules.install")
+
     on_config(function (target)
         -- we disable to build across targets in parallel, because the source files may depend on other target modules
         -- @see https://github.com/xmake-io/xmake/issues/1858
-        local target_with_modules
-        for _, dep in ipairs(target:orderdeps()) do
-            local sourcebatches = dep:sourcebatches()
-            if sourcebatches and sourcebatches["c++.build.modules"] then
-                target_with_modules = true
-                break
-            end
-        end
+        local common = import("modules_support.common")
+
+        local target_with_modules = target:sourcebatches()["c++.build.modules"] and common.contains_modules(target) or false
```

Comment:
we can improve `contains_modules ` for `target:sourcebatches()["c++.build.modules"]`

just 

```lua
        local target_with_modules = common.contains_modules(target)

```